core/store/src/trie/receipts_column_helper.rs

```diff
@@ -232,6 +232,35 @@ impl ShardsOutgoingReceiptBuffer {
     fn write_indices(&self, state_update: &mut TrieUpdate) {
         set(state_update, TrieKey::BufferedReceiptIndices, &self.shards_indices);
     }
+
+    /// Fetch the first receipt from the buffer.
+    /// Can't implement using OutgoingReceiptBuffer::iter() because it requires a &mut reference.
+    pub fn get_first_receipt(
+        &self,
+        to_shard: ShardId,
+        trie: &dyn TrieAccess,
+        side_effects: bool,
+    ) -> Result<Option<ReceiptOrStateStoredReceipt>, StorageError> {
```

Comment:
Does this method actually remove the receipt from the buffer? Does it clone it or will it be read again later? Do we care?

Reply:
nit: maybe call it peek_receipt?

Reply:
It doesn't remove anything. In Rust `first()` and `get()` are usually methods that access a collection without removing the item, so I thought that this would be similar.
I can change it to peek if it's confusing